Sources/SwiftParser/StringLiterals.swift

```diff
@@ -595,10 +595,80 @@ extension Parser {
       )
     }
   }
+
+  mutating func parseSimpleString() -> RawSimpleStringLiteralExprSyntax {
+    let openDelimiter = self.consume(if: .rawStringDelimiter)
+    let (unexpectedBeforeOpenQuote, openQuote) = self.expect(anyIn: SimpleStringLiteralExprSyntax.OpenQuoteOptions.self, default: .stringQuote)
+
+    /// Parse segments.
+    var segments: [RawStringSegmentSyntax] = []
+    var loopProgress = LoopProgressCondition()
+    while loopProgress.evaluate(self.currentToken) {
+      // If we encounter a token with leading trivia, we're no longer in the
+      // string literal.
+      guard currentToken.leadingTriviaText.isEmpty else { break }
+
+      if let stringSegment = self.consume(if: .stringSegment, TokenSpec(.identifier, remapping: .stringSegment)) {
+        var unexpectedAfterContent: RawUnexpectedNodesSyntax?
+
+        if let backslash = self.consume(if: .backslash) {
+          var tokens: [RawTokenSyntax] = [backslash]
+
+          var interpolationLoopProgress = LoopProgressCondition()
+          while interpolationLoopProgress.evaluate(self.currentToken) {
+            if let rightParen = self.consume(if: TokenSpec.rightParen) {
+              tokens += [rightParen]
+              break
+            } else {
+              tokens += [self.consumeAnyToken()]
+            }
+          }
+
+          unexpectedAfterContent = RawUnexpectedNodesSyntax(
+            tokens,
+            arena: self.arena
+          )
+        }
+
+        segments.append(RawStringSegmentSyntax(content: stringSegment, unexpectedAfterContent, arena: self.arena))
+      } else {
+        break
+      }
+    }
+
+    let (unexpectedBetweenSegmentAndCloseQuote, closeQuote) = self.expect(
+      anyIn: SimpleStringLiteralExprSyntax.CloseQuoteOptions.self,
+      default: openQuote.closeTokenKind
+    )
+    let closeDelimiter = self.consume(if: .rawStringDelimiter)
+
+    return RawSimpleStringLiteralExprSyntax(
+      RawUnexpectedNodesSyntax(combining: unexpectedBeforeOpenQuote, openDelimiter, arena: self.arena),
+      openQuote: openQuote,
+      segments: RawSimpleStringLiteralSegmentsSyntax(elements: segments, arena: self.arena),
+      unexpectedBetweenSegmentAndCloseQuote,
+      closeQuote: closeQuote,
+      RawUnexpectedNodesSyntax([closeDelimiter], arena: self.arena),
+      arena: self.arena
+    )
+  }
 }
 
 // MARK: - Utilities
 
+fileprivate extension RawTokenSyntax {
+  var closeTokenKind: SimpleStringLiteralExprSyntax.CloseQuoteOptions {
+    switch self {
+    case .multilineStringQuote:
+      return .multilineStringQuote
+    case .stringQuote:
+      return .stringQuote
+    default:
+      fatalError("Unsupported type")
```

Comment:
I hate seing `fatalError` in the parser unless it’s super obvious why it can’t happen. What do you think about defaulting this to `.stringQuote` here? That does seem like a reasonable default to me.